### DISCUSSION TITLE
An option to play a playlist in reverse order

### DISCUSSION THREAD

**Discussion by swishkin** [navidrome/navidrome#2345] [Ideas]
In Spotify, there are a number of columns in a playlist by default, all of which we can sort the playlist by; Title, Album, Date Added, and Duration. When sorted by any one of these, instead of the default custom order, the playlist will play in the sorted order rather than in the default custom order.

Navidrome's behavior, in my experience, is to always play the playlist from start to end, no matter how the playlist is sorted. If the playlist is sorted in reverse by the index number of the track, it still plays the songs in ascending order.

Playing a playlist from the beginning will play all the songs in the playlist in the default custom order:

![image](https://user-images.githubusercontent.com/19404743/236931462-05725d3d-7903-4341-b144-5061fbc9f9e9.png)

Playing a playlist from the end, but with the playlist sorted in reverse order, will play only the last song, then stop, as it is still playing the playlist in its original order, not respecting the current view/sort:

![image](https://user-images.githubusercontent.com/19404743/236931734-9783e20c-542d-4193-8587-a1bc7b33d8bd.png)

Instead, I suggest Navidrome should play in the order sorted, at _least_ for the track number column, if not all available columns; notice that the track number would decrement when the next track played instead of incrementing.

![image](https://user-images.githubusercontent.com/19404743/236932496-cad6c1d1-42b5-46ee-98d0-60286c7e3f51.png)

Please let me know if there's a way to do this that I'm not aware of, otherwise I hope this feature request/idea is clear.

**Comment by deluan** [comment]
This seems to be a bug. I'll take a look later and convert this issue to a bug report if this is the case. Thanks for reporting!

**Reply by swishkin** [reply]
Thanks for replying! Is there any progress here, or anything I can do to help?

**Reply by deluan** [reply]
I can confirm this is a bug, but I'm not fixing it in the current UI implementation. I rather work on the new UI, that will solve some of the issues we have with playlists being different than albums, which causes some bugs and issues like this.
